dali/operators/expressions/arithmetic.h

```diff
@@ -244,9 +247,10 @@ class ArithmeticGenericOp : public Operator<Backend> {
  private:
   void AllocateIntermediateNodes() {
     auto &expr = *expr_;
-    bool is_simple_expression =
-        expr.GetNodeType() == NodeType::Function && expr.GetSubexpressionCount() == 2;
-    auto &func = dynamic_cast<ExprFunc&>(expr);
+    bool is_simple_expression = expr.GetNodeType() == NodeType::Function &&
+                                expr.GetSubexpressionCount() > 0 &&
+                                expr.GetSubexpressionCount() <= 2;
+    auto &func = dynamic_cast<ExprFunc &>(expr);
     for (int i = 0; i < func.GetSubexpressionCount(); i++) {
       is_simple_expression = is_simple_expression && func[i].GetNodeType() != NodeType::Function;
     }
```

Comment:
Remove "binary" from the error message below.

Reply:
Done